src/bootstrap/native.rs

```diff
@@ -576,6 +564,18 @@ fn configure_cmake(
         ldflags.push_all(&flags);
     }
 
+    // For distribution we want the LLVM tools to be *statically* linked to libstdc++.
+    // We also do this if the user explicitly requested static libstdc++.
+    if builder.config.llvm_tools_enabled || builder.config.llvm_static_stdcpp {
```

Comment:
We blanket enable llvm-static-stdcpp in our CI (see https://github.com/rust-lang/rust/blob/03918badd33d255de806b4a9a8aa75b031ed0738/src/ci/run.sh#L72), so I'm wondering if it makes sense to remove the llvm-tools enablement here -- it's not obvious to me that it *should* be present.

It looks like that was originally added in https://github.com/rust-lang/rust/pull/50336, but it seems like an odd thing to force opt-in users of the build system to with an orthogonal flag. Maybe you have a better sense here though!

I think this somewhat makes this PR a clearer "yes" to me, since if we're migrating to a simpler story of only enabling static stdcpp if asked explicitly, then doing that for everything makes sense. But if that's gated on llvm tools being enabled, it feels weird for that to apply to "not tools".

Reply:
It's a tough call. When you need `llvm-tools`, you'll very often want them with static libstdc++, because otherwise they may just not run on whatever host you try running them on, and you won't be able to tell until a user with an old host is bitten by it. But at the same time, as you point out, there's no _fundamental_ reason they need to be coupled (at least as far as I can tell), which suggests the two flags should be decoupled. I think of this as an anti-footgun of having `llvm-tools` imply `static-libstdc++` because builders may not have thought through the implications of _not_ linking it statically (they may not even be aware of the `static-stdcpp` flag).

Which is all to say — I think decoupling them gives more flexibility to builders, but at the cost of making it harder to get to a functioning build for someone who's trying to replicate the upstream build process. Which of those is more important, I'm not sure.

Reply:
Hm. Yes, that makes sense. I think we should probably be changing the default for static stdcpp, then, rather than specifically checking tools here, but that seems like a change that needs it's own PR.

Reply:
Oh, yeah, I like that idea! Can submit a PR for that tomorrow. And then you'd want this PR to _just_ switch on `llvm_static_stdcpp`?

Reply:
Actually, I think we should do a separate PR that changes the `if` condition to just be `llvm_static_stdcpp` _and_ flips its default. Otherwise, in the interrim (however short) we'll be changing the behavior of the code.